### PR TITLE
PR: Profile picture and upload files component

### PR DESCRIPTION
**Pull request by tissakhosla**
Created user upload component for login pages.

### CODE REVIEW

File: client/src/components/Button/Button.css [hunk 1]
```diff
@@ -20,6 +21,10 @@
   padding: 0.13em 2.71em;
   font-size: 36px;
 }
+.Button--extra-small {
+  padding: 0.30em 2.35em;
+  font-size: 15px;
+}
 .Button--small {
   padding: 0.42em 3.35em;
```
Comment (ZakkMan) [inline client/src/components/Button/Button.css]: Did you check in with whoever built this out before making these changes?

File: client/src/components/ProfilePicture/NewUserUploadPicture.css [hunk 2]
```diff
@@ -0,0 +1,22 @@
+.NewUserUpload {
+  height: 233px;
+  width: 233px;
+  border: solid var(--yellow) 1px;
+  border-radius: 50%;
+  text-align: center;
+  /* display: flex;
+  flex-direction: column;
+  justify-content: center; */
```
Comment (ZakkMan) [inline client/src/components/ProfilePicture/NewUserUploadPicture.css]: Delete commented out code! Never commit commented out code!

File: client/src/components/ProfilePicture/NewUserUploadPicture.css [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+}
+
+.NewUserUpload__p {
```
Comment (ZakkMan) [inline client/src/components/ProfilePicture/NewUserUploadPicture.css]: If this is just a `p` tag, then just use the descendent selector: `.NewUserUpload p`

File: client/src/components/ProfilePicture/NewUserUploadPicture.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import React from "react";
+import './NewUserUploadPicture.css'
+import Button from '../Button/Button'
+
+const handleClick = e => {
+  console.log("Clicked");
+}; //from ../Button/Button.stories.js
+
```
Comment (ZakkMan) [inline client/src/components/ProfilePicture/NewUserUploadPicture.js]: This function should be passed down as a prop

**Comment by erinkelley27** [comment]
@tissakhosla Is there an issue for this?

File: client/src/components/ProfilePicture/ProfilePicture.stories.js [hunk 5]
```diff
@@ -1,14 +1,18 @@
...
   .add("Medium with Link", () => <ProfilePicture to="#" image={Image} />)
   .add("Large with Link", () => <ProfilePicture type="large" to="#" image={Image} />)
-  .add("Extra-Small Static", () => <ProfilePicture type="extra-small" image={Image} />)
+  .add("Extra Small Static", () => <ProfilePicture type="extra-small" image={Image} />)
   .add("Small Static", () => <ProfilePicture type="small" image={Image} />)
   .add("Medium Static", () => <ProfilePicture image={Image} />)
   .add("Large Static", () => <ProfilePicture type="large" image={Image} />);
+
+storiesOf("New User Upload", module)
+  .add("New User Upload", () => <NewUserUpload></NewUserUpload>)
```
Comment (ZakkMan) [inline client/src/components/ProfilePicture/ProfilePicture.stories.js]: Why is this here and not in the `NewUserUpload.stories.js` file?

File: client/src/components/ProfilePicture/NewUserUploadPicture.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+const NewUserUpload = () => {
```
Comment (ZakkMan) [inline client/src/components/ProfilePicture/NewUserUploadPicture.js]: `NewUser` seems like contextual information - there may be places where we want the user to upload something that has nothing to do with creating a new user profile. Maybe we could call this `FileUpload`, or something like that?

File: client/src/components/Button/Button.css (hunk 1 above)
Comment (ZakkMan) [reply client/src/components/Button/Button.css]: @tissakhosla

File: client/src/components/FileUploadComponent/FileUpload.js [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+
+    } else {
+      this.setState({
+        renderThis:
+          <div onChange={this.handleChange}>
+            <label for="inputTypeFile" className="FileUpload__label" >{this.props.buttonLabel}</label>
+            <input type="file" id="inputTypeFile" className='FileUpload' ></input>
+          </div>
+      })
+    }
```
Comment (ZakkMan) [inline client/src/components/FileUploadComponent/FileUpload.js]: Can you put all this render code into the `render` method and remove the conditional?

File: client/src/components/FileUploadComponent/FileUpload.js [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+
+  handleChange(eo) {
+    eo.preventDefault()
+    eo.persist()
+    let files = eo.target.files
+    console.warn("data files", files)
+    let reader = new FileReader();
+    reader.readAsDataURL(files[0]);
+    reader.onload = (eo) => {
+      console.warn("img data ", eo.target.result)
```
Comment (ZakkMan) [inline client/src/components/FileUploadComponent/FileUpload.js]: Remove any `console.log` statements

File: client/src/components/FileUploadComponent/FileUpload.js [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+
+    this.state = {
+      renderThis: null,
+      style: null
+    }
+
+    this.handleChange = this.handleChange.bind(this)
+  }
+
+  handleChange(eo) {
```
Comment (ZakkMan) [inline client/src/components/FileUploadComponent/FileUpload.js]: Can you call `eo` `evt`

File: client/src/components/FileUploadComponent/FileUpload.css [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+  border: solid rgba(0, 0, 0, 0);
+  border-radius: 5px;
+  font-family: var(--default-font);
+  padding: 0.30em 2.35em;
+  font-size: 15px;
+  color: var(--white);
+  background-color: var(--pink);
+}
+
+.NewUserUpload {
```
Comment (ZakkMan) [inline client/src/components/FileUploadComponent/FileUpload.css]: Why is this class called `NewUserUpload` and not something related to `FileUpload`?

File: client/src/components/FileUploadComponent/FileUpload.stories.js [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+
+
+storiesOf("File Upload", module)
+  .add("Default", () =>
+    <FileUpload>
+    </FileUpload>)
+  .add("New User Upload", () =>
+    <FileUpload
+      context="userPhoto"
+      buttonLabel="Browse Files">
```
Comment (ZakkMan) [inline client/src/components/FileUploadComponent/FileUpload.stories.js]: I think you'll need more props than this. We'll want a way to control the file type through a prop, a way to control the header text, the button label, and a callback that can accept the file and do something with it
